Add comparison with k3s to docs (#738)
eclipse-ankaios/ankaios: doc/docs/reference/orch-comparison.md

> # Comparing Container Orchestrators

Reviewer: ```suggestion
# Comparing container orchestrators
```

See documentation [guideline](https://eclipse-ankaios.github.io/ankaios/latest/development/documentation-guidelines/#capitalization).

> This page presents a comparison between Eclipse Ankaios and the Lightweight Kubernetes distribution K3s. All benchmarks were run on identical virtual machines in a controlled environment.

Reviewer: ```suggestion
This page presents a comparison between Eclipse Ankaios and the lightweight Kubernetes distribution K3s. All benchmarks were run on identical virtual machines in a controlled environment.
```

See documentation [guideline](https://eclipse-ankaios.github.io/ankaios/latest/development/documentation-guidelines/#capitalization).

> This page presents a practical, measured comparison between Eclipse Ankaios and Lightweight Kubernetes (K3s). All benchmarks were run on identical virtual machines in a controlled environment.

Reviewer 2: ```suggestion
This page presents a comparison between Eclipse Ankaios and the Lightweight Kubernetes distribution K3s. All benchmarks were run on identical virtual machines in a controlled environment.
```

> ## Test Environment

Reviewer: ```suggestion
## Test environment
```

> ### Host Machine

Reviewer: ```suggestion
### Host machine
```

| Property     | Value                                                  |
|--------------|--------------------------------------------------------|
| **CPU**      | Intel Core i7-11850H @ 2.50 GHz (8 cores / 16 threads) |
| **RAM**      | 16 GiB                                                 |
| **OS**       | Ubuntu 24.04 via WSL2 (Windows)                        |

Virtual machines are managed with [Incus](https://linuxcontainers.org/incus/) on the host.

> ### Virtual Machine Configuration

Reviewer: ```suggestion
### Virtual machine configuration
```

| Property     | Value                |
|--------------|----------------------|
| **VM image** | Ubuntu 24.04 (noble) |
| **vCPUs**    | 4                    |
| **RAM**      | 4 GiB                |
| **Disk**     | 20 GiB               |

> One VM was used per orchestrator, reset to a clean snapshot before each measurement. No workloads were running during idle measurements. Benchmark images were pre-pulled into the snapshot so that no registry pulls occur during runs.

Reviewer 2: ```suggestion
One VM was used per orchestrator, reset to a clean snapshot before each measurement. No workloads were running during idle measurements. Benchmark images were pre-pulled into the snapshot so that no registry pulls occur during runs and the measurements are not influenced by network latency or bandwidth fluctuations.
```

### Orchestrators used

| Property      | Eclipse Ankaios    | K3s                  |
|---------------|--------------------|----------------------|
| **Version**   | v1.0.0             | v1.35.4+k3s1         |

> | **Developer** | Eclipse Foundation | Rancher (SUSE)       |

Reviewer: ```suggestion
| **License** | Apache-2.0 |  Apache-2.0       |
```

The "Eclipse Foundation" is not the "developer" of Ankaios. Eclipse provides rather the governance framework. I would replace this with license.

| **Runtime** | Podman          | containerd (bundled) |

---

> ## Feature Comparison

Reviewer: ```suggestion
## Feature comparison
```

| Dimension                     | Ankaios                            | K3s            |
|-------------------------------|------------------------------------|----------------|
| **Focus**                     | Automotive / embedded (SDV)        | Edge / IoT     |
| **Language**                  | Rust                               | Go             |
| **Workload types**            | OCI containers (multiple runtimes) | OCI containers |
| **Multi-node orchestrator**   | ✔️                                 | ✔️             |
| **Reconciliation loop**       | ✔️                                 | ✔️             |
| **API access for workloads**  | ✔️                                 | ✔️             |
| **Supports dependencies**     | ✔️                                 | ❌             |
| **Flash wear-out prevention** | ✔️                                 | ❌             |
| **Requirement tracing**       | ✔️                                 | ❌             |

---

> ## Benchmark Results

Reviewer: ```suggestion
## Benchmark results
```

Each benchmark was run 100 times per orchestrator; results below are averages. Memory figures use PSS (Proportional Set Size) from `/proc/<pid>/smaps_rollup` rather than RSS to avoid double-counting shared pages.

---

> ### Startup Time

Reviewer: ```suggestion
### Startup time
```

Startup time measures how long each orchestrator takes from process launch until its management API is responsive.

<img src="../../assets/comp-startup-time.png" width="700" alt="Startup time" />

---

### Resource usage in idle mode

Resource usage in idle mode is measured after each orchestrator has started and stabilized, with no workloads running.

<img src="../../assets/comp-idle-resources.png" width="700" alt="Idle resource usage" />

---

> ### Workload Deployment Time

Reviewer: ```suggestion
### Workload deployment time
```

> This scenario submits an nginx manifest and measures the time until the first HTTP 200 response, the true end-to-end deployment time.

Reviewer 2: Maybe we can also mention the sampling rate at which the measurement was done.

Author: Just added it.

Author: Mentioned the sampling time in the "Benchmark results" section, so that it covers all scenarios which use polling.

<img src="../../assets/comp-deploy-nginx.png" width="700" alt="Deploy nginx to HTTP 200" />

---

> ### Rolling Update Time

Reviewer: ```suggestion
### Rolling update time
```
